test/javascripts/components/d-editor-test.js.es6

```diff
@@ -193,6 +193,23 @@ testCase('link modal (simple link)', function(assert) {
   });
 });
 
+testCase('link modal (simple link) with selected text', function(assert, textarea) {
+  this.set('value', "hello world.");
```

Comment:
https://github.com/cpradio/discourse/blob/cbde168281e2c7a37a0a4701677d8a0e0ec32344/test/javascripts/components/d-editor-test.js.es6#L98-L101

This uses andThen to select text...